magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from "react";
+import browser from "browser-detect";
+import { config } from "../../config";
+import "./DataPreviewMapOpenInNationalMapButton.scss";
+import { BrowserDetectInfo } from "browser-detect/dist/types/browser-detect.interface";
+import { ParsedDistribution } from "helpers/record";
+
+type PropsType = {
+    distribution: ParsedDistribution;
+    buttonText: string;
+    style: {
+        [key: string]: any;
+    };
+};
+
+const DEFAULT_TARGET_URL = "https://nationalmap.gov.au/";
+
+class DataPreviewMapOpenInNationalMapButton extends Component<PropsType> {
+    private browser: BrowserDetectInfo;
+    private winRef: Window | null;
+
+    constructor(props) {
+        super(props);
+        this.state = {};
+        this.onPopUpMessageReceived = this.onPopUpMessageReceived.bind(this);
+        this.winRef = null;
+        this.browser = browser();
+    }
+
+    componentDidMount() {
+        if (this.browser.name === "ie" && this.browser?.versionNumber! < 12)
+            return;
+        window.addEventListener("message", this.onPopUpMessageReceived);
+    }
+
+    componentWillUnmount() {
+        if (this.browser.name === "ie" && this.browser?.versionNumber! < 12)
+            return;
+        window.removeEventListener("message", this.onPopUpMessageReceived);
+    }
+
+    createCatalogItemFromDistribution(withoutBaseMap = false) {
+        const { distribution } = this.props;
+        let catConfig;
+
+        if (config?.supportExternalTerriaMapV7 === true) {
+            catConfig = {
+                initSources: [
+                    {
+                        catalog: [
+                            {
+                                name: distribution?.title,
+                                type: "magda-item",
+                                distributionId: distribution?.identifier,
+                                url: config.baseExternalUrl,
+                                isEnabled: true,
+                                zoomOnEnable: true
+                            }
+                        ]
+                    }
+                ]
+            };
+        } else {
+            const dga_id_prefix = "data.gov.au-postMessage-";
+            catConfig = {
+                initSources: [
+                    {
+                        catalog: [
+                            {
+                                name: distribution?.title,
+                                type: "magda",
+                                recordId: distribution?.identifier,
+                                url: config.baseExternalUrl,
+                                id:
+                                    dga_id_prefix +
+                                    this.props.distribution.identifier
```

Comment:
Minor improvement:
const identifier= dga_id_prefix + distribution?.identifier;
then use it in both catalog and workbench to avoid duplicates.

Reply:
Good idea~ Updated